testing/sdk_tasks.py

```diff
@@ -60,8 +60,9 @@ def fn():
 
         old_set = set(old_task_ids)
         new_set = set(task_ids)
-        newly_launched_set = (new_set - old_set)
-        all_updated = len(newly_launched_set) == len(new_set)
+        newly_launched_set = new_set.difference(old_set)
+        old_remaining_set = old_set.intersection(new_set)
+        all_updated = len(newly_launched_set) == len(new_set) and len(old_remaining_set) == 0
```

Comment:
Should we not also check that the new set has the same length as the old set? If we had (t1, t2) and then launch (t3) these checks will pass?

Would:

 len(new_set) == len(old_set) and len(old_remaining_set)

Not be the correct condition?

Sorry for the formatting, I'm on my phone.

Reply:
@elezar the condition implied by all_updated is that there are no old tasks present, not that the new set matches the old set. To translate the condition as all old tasks have a corresponding new task would fail if a plan launched multiple tasks where previously only a single task was launched (or the inverse).

The follow on condition check that the plan completed covers the overarching concern, that the plan completed.

Reply:
OK. Thanks for the clarification. I understood it differently, and think the original implementation wasn't implemented this way either:
```python
        all_updated = True
        for id in task_ids:
            if id in old_task_ids:
                all_updated = False
        if len(task_ids) < len(old_task_ids):
            all_updated = False
        return all_updated
```
I'm not saying that this was correct for HDFS, but changing this does break the hello-world tests.

This may mean that we need two checks:
- `tasks_replaced`: Checks for 1-1 replacement of tasks
- `tasks_removed_or_updated`: Checks that the intersection of the new task list and the old task list is zero.

For what it's worth, it may be useful move the inner function to the module level so that we could exercise it with a couple of unit tests, but that may be out of scope of this PR.